ui/app/contexts/metametrics.new.js

```diff
@@ -92,16 +92,27 @@ export function MetaMetricsProvider ({ children }) {
       const idTrait = metaMetricsId ? 'userId' : 'anonymousId'
       const idValue = metaMetricsId ?? METAMETRICS_ANONYMOUS_ID
       const match = matchPath(location.pathname, { path: PATHS_TO_CHECK, exact: true, strict: true })
-      if (
-        match &&
+      // Start by checking for a missing match route. If this falls through to the else if, then we know we
+      // have a matched route for tracking.
+      if (!match) {
+        // We have more specific pages for each type of transaction confirmation
+        // The user lands on /confirm-transaction first, then is redirected based on
+        // the contents of state.
+        if (location.pathname !== '/confirm-transaction') {
+          // Otherwise we are legitimately missing a matching route
+          captureMessage(`Segment page tracking found unmatched route`, {
+            level: Severity.Info,
+            previousMatch,
+            currentPath: location.pathname,
```

Comment:
The Sentry context only allows specific keys (at least that's my interpretation of their docs: https://docs.sentry.io/platforms/javascript/enriching-events/context/#passing-context-directly)

I think we want these to be searchable, so I'd set them as tags:
```suggestion
            tags: {
              previousMatch,
              currentPath: location.pathname,
            },
```

Reply:
https://docs.sentry.io/platforms/javascript/enriching-events/context/#extra-data-additional-data here it seems like any extra keys passed will be added to context. I can make these tags just want to make sure this is how we want the data represented in sentry. Tags seem more like things that can be attributed to many events, like browser version, etc... Again. I'm fine either way :)

Reply:
Fair enough! Though we can't search the `extra` data, so there will be no way to distinguish between different pages on Sentry. They'll all get grouped under one issue.

Hmm :thinking:

Reply:
Well, I guess a lack of indexing is fine if we never expect to get this message. I think I agree that it's better for these to be in the extra data for now. We can always revisit this if it becomes an issue.